pypdf/_writer.py

```diff
@@ -1115,18 +1181,29 @@ def clone_reader_document_root(self, reader: PdfReader) -> None:
         Args:
             reader: PdfReader from which the document root should be copied.
         """
-        self._objects.clear()
+        if self.incremental:
+            self._objects = [None] * cast(int, reader.trailer["/Size"])
+        else:
+            self._objects.clear()
         self._root_object = reader.root_object.clone(self)
-        self._root = self._root_object.indirect_reference  # type: ignore[assignment]
         self._pages = self._root_object.raw_get("/Pages")
+
+        assert len(self._objects) <= cast(int, reader.trailer["/Size"])  # for pytest
```

Comment:
How does this affect *pytest*?

Reply:
it is here to detect possible regressions